src/main.rs

```diff
@@ -122,16 +129,38 @@ fn settings() -> io::Result<Settings> {
         .get_matches();
 
     // Read the threshold.
-    let default_threshold = Byte::from_str(DEFAULT_THRESHOLD).unwrap(); // Manually verified safe
+    let default_threshold = Threshold::Absolute(
+        Byte::from_str(DEFAULT_THRESHOLD).unwrap(), /*  Manually verified safe */
+    );
     let threshold = matches.value_of(THRESHOLD_OPTION).map_or_else(
         || Ok(default_threshold),
-        |threshold| {
-            Byte::from_str(threshold).map_err(|_| {
-                io::Error::new(
-                    io::ErrorKind::InvalidInput,
-                    format!("Invalid threshold {}.", threshold.code_str()),
-                )
-            })
+        |threshold| match threshold.strip_suffix('%') {
+            Some(threshold_percentage_string) => {
+                // Threshold parameter has "%" suffix: Try parsing as f64
+                threshold_percentage_string
+                    .trim()
+                    .parse::<f64>()
+                    .map_err(|parse_error| {
+                        io::Error::new(
+                            io::ErrorKind::InvalidInput,
+                            format!(
+                                "Invalid relative threshold {}. Error: {}",
+                                threshold.code_str(),
+                                parse_error,
```

Comment:
Since we don't include the parse error in the absolute case, maybe we should omit it here too? It's not like we're parsing some complex struct where you need to figure out which field was problematic.

In my experience, Rust throws errors with inconsistent capitalization and punctuation, so I often prefer to have full control over error messages and only propagate underlying errors when not doing so would make debugging difficult (e.g., IO error messages should generally be surfaced verbatim since a billion things can go wrong with IO).

(But most of all I just want to be consistent.)